### ISSUE TITLE
Special url characters not used in Soql

### ISSUE THREAD

**Issue by wpeereboom** [codelicia/trineforce#27]
When parameter contains url special character it's ignored in Soql:
`$phone = '+31612345678';

$sql = $conn->createQueryBuilder()
    ->select(['Contact.name','Contact.phone','Account.id', 'Account.name', 'Owner.email', 'Owner.name', 'Owner.id'])
    ->from('Contact')
    ->where('Phone = :phone')
    ->setParameter('phone', $phone, 'string')
    ->setMaxResults(1)
    ->execute();`

No results.

With urlencode in `do_fetch`
`
/** @return mixed[]|false */
    private function doFetch()
    {
        // TODO: how to deal with different versions? Maybe `driverOptions`?
        $request = $this->connection->get('/services/data/v20.0/query?q=' . urlencode($this->statement));

        return json_decode($request->getBody()->getContents(), true);
    }
`

One result

**Comment by malukenho** [comment]
I currently have the same issue, the work-around would be something similar to:

```php
if (in_array('PHONE NUMBER', $duplicationTypes)) {
    $stmt->orWhere('Phone = :Phone')
        ->setParameter('Phone', urlencode('+31639497661'), ParameterType::STRING);
}
```

**Comment by malukenho** [comment]
@all-contributors please add @wpeereboom for bug

**Comment by malukenho** [comment]
@all-contributors please add @wpeereboom for bug

**Comment by malukenho** [comment]
Fixed on #28
